### ISSUE TITLE
Errors are Swallowed in Callback function

### ISSUE THREAD

**Issue by zcr1** [darrachequesne/socket.io-msgpack-parser#1]
Whenever I enable the custom parser, any errors in the callback function chain are swallowed. I can reproduce with the following code on my project in a dummy class.
```
socket.on('connect', this.foo.bind(this))
foo(){
  throw 'foo';
}

```

**Comment by darrachequesne** [comment]
@zcr1 hmm.. what is the behaviour with the default parser?

**Comment by zcr1** [comment]
The error is shown in the console.

**Comment by darrachequesne** [comment]
Hi! For future readers, it should be fixed by https://github.com/darrachequesne/socket.io-msgpack-parser/commit/388e5f55862e32562bbfb0e6e1066dc3f2962191 (included in version `2.2.1`).

Sorry for the lack of attention...
